plugins.d/Lets_Encrypt/dns_01.py

```diff
@@ -3,73 +3,197 @@
+from os import makedirs, chmod, chown
+from os.path import isfile, join, exists, dirname, realpath
+from shutil import which, copy
 
-LEXICON_CONF = '/usr/share/confconsole/letsencrypt/lexicon.yml'
+from typing import Optional
 
-LEXICON_CONF_NOTE = '''# Configure according to lexicon documentation https://dns-lexicon.readthedocs.io/
-# Note that documentation began around v.3.3.28 of lexicon. Therefore not all of
-# the features might be available to you!
-'''
+LEXICON_SHARE_DIR = '/usr/share/confconsole/letsencrypt'
+LEXICON_CONF_DIR = '/etc/dehydrated'
 
-LEXICON_CONF_MAX_LINES = 7
+#LEXICON_CONF_MAX_LINES = 7
 
-
-def load_config():
-    ''' Loads lexicon config if present '''
-    config = []
-    if not path.isfile(LEXICON_CONF):
-        while len(config) < LEXICON_CONF_MAX_LINES:
-            config.append('')
-        return config
+def load_config(provider: str) -> tuple[str, list[str]]:
+    ''' Loads lexicon config if present, loads example if not,
+    returns tuple(conf_file, list(config))
+    '''
+    if provider in ['route53', 'cloudflare']:
+        example_conf = join(LEXICON_SHARE_DIR,
+                            f'lexicon-confconsole-provider_{provider}.yml')
     else:
-        with open(LEXICON_CONF, 'r') as fob:
-            for line in fob:
-                line = line.rstrip()
-                if line and not line.startswith('#'):
-                    config.append(line)
-
-        while len(config) > LEXICON_CONF_MAX_LINES:
-            config.pop()
-        while len(config) < LEXICON_CONF_MAX_LINES:
-            config.append('')
-        return config
-
-
-def save_config(config):
+        example_conf = join(LEXICON_SHARE_DIR,
+                            'lexicon-confconsole-provider_example.yml')
+    conf_file = join(LEXICON_CONF_DIR, f'lexicon_{provider}.yml')
+    config = []
+    if not isfile(conf_file):
+        copy(example_conf, conf_file)
+        # ensure root read/write only as file contains sensitive info
+        chown(conf_file, 0, 0) # chown root:root
+        chmod(conf_file, 0o600) # chmod 600 (owner read/write only)
+    with open(conf_file, 'r') as fob:
+        for line in fob:
+            config.append(line.rstrip())
+
+    #    while len(config) > LEXICON_CONF_MAX_LINES:
+    #       config.pop()
+    #    while len(config) < LEXICON_CONF_MAX_LINES:
+    #        config.append('')
+    return conf_file, config
+
+
+def save_config(conf_file: str, config: str) -> None:
     ''' Saves lexicon configuration '''
-    with open(LEXICON_CONF, 'w') as fob:
-        fob.write(LEXICON_CONF_NOTE)
+    with open(conf_file, 'w') as fob:
         for line in config:
-            line = line.rstrip()
             if line:
-                fob.write(line + '\n')
-
+                fob.write(line.rstrip() + '\n')
+    # ensure root read/write only as file contains sensitive info
+    chown(conf_file, 0, 0) # chown root:root
+    chmod(conf_file, 0o600) # chmod 600 (owner read/write only)
```

Comment:
Looks like this takes a string and writes each character to a new line, is this on purpose?